### ISSUE TITLE
`random-route` creates a new route on every `cf push` when `domain` is also specified

### ISSUE THREAD

**Issue by flavorjones** [cloudfoundry/cli#1177]
## Command  

Run this command twice:

`cf push my-app --random-route -d my-domain.com`

## What occurred

You'll see that *two* random routes are associated with the application.

## What you expected to occur

I would expect that a random route would be created on the *first* push, and that the second push would re-use that route.

This is indeed the behavior we see if we run the same command without specifying domain:

`cf push my-app --random-route`

## CLI Version  

```
$ cf -v
cf version 6.28.0+9e024bd.2017-06-27
```

## CC API Endpoint Version  

```
$ cf api
api endpoint:   https://api.run.pivotal.io
api version:    2.84.0
```

## Platform & Shell Details  

```
$ cat /etc/lsb-release 
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=18
DISTRIB_CODENAME=sarah
DISTRIB_DESCRIPTION="Linux Mint 18 Sarah"

$ uname -a
Linux curiosity 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

$ bash --version
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

## Any other relevant information  

Note that this occurs whether the options are specified as commandline parameters, or in an application manifest.

Also note that the workaround of "don't specify a domain" isn't available for TCP routes, which require a domain to be specified.

Tagging @joshuamckenty who reported this.

**Comment by dkoper** [comment]
Thanks! I agree this is unexpected - we'll take a look.

**Comment by dkoper** [comment]
This is now resolved in cf CLI 6.34.0.

Regards,
Dies Koper
CF CLI PM
